ClimsoftVer4/ClimsoftVer4/ucrValueFlagPeriod.vb

```diff
@@ -221,13 +233,23 @@ Public Class ucrValueFlagPeriod
 
     End Sub
 
-    Private Sub ucrValue_Leave(sender As Object, e As EventArgs) Handles ucrValue.Leave
+    Private Sub ucrValue_evtValueChanged(sender As Object, e As EventArgs) Handles ucrValue.evtValueChanged
+
         QCForValue()
+
+        'Remove the missing value flag for non empty
+        If Not ucrValue.IsEmpty AndAlso ucrFlag.GetValue = "M" Then
+            ucrFlag.SetValue("")
+        End If
```

Comment:
@Patowhiz  DO we need to do this twice on `evtValueChanged `and  on this if statement above `If sender Is ucrValue.txtBox Then`

Reply:
On enter, we want to set the flag as an M if the value textbox is empty.
On value changed we want to remove the M in the flag, if there is any, when the value is not empty anymore because M stand missing value.

So technically, the other one adds, this one removes. I separated it because from our discussions we don't want the M to be added on event leave, we only want it added on enter. Following this, event leave of all textboxes always raises value changed.

Reply:
Okay sounds good then. (y)